dDatabase/CDClientDatabase/CDClientTables/CDDeletionRestrictionsTable.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "CDTable.h"
+
+enum class eDeletionRestrictionsCheckType : uint32_t;
+
+struct CDDeletionRestriction {
+	uint32_t id;
+	bool restricted;
+	std::vector<uint32_t> ids;
+	eDeletionRestrictionsCheckType checkType;
+};
+
+class CDDeletionRestrictionsTable : public CDTable<CDDeletionRestrictionsTable, std::map<uint32_t, CDDeletionRestriction>> {
+public:
+	void LoadValuesFromDatabase();
+	const CDDeletionRestriction& GetByID(uint32_t id);
+
+	static CDDeletionRestriction Default;
```

Comment:
It would be great if this could be not public, and instead use a getter which returns a const reference to a default struct so this cannot be modified accidentally, alongside removing a static variable in place for a static function and an unnamed namespace member.

Reply:
Tell that to itemComponentTable